src/main/java/TypewiseAlert/TypewiseAlert.java

```diff
@@ -1,87 +1,47 @@
 package TypewiseAlert;
 
-public class TypewiseAlert 
-{
-    public enum BreachType {
-      NORMAL,
-      TOO_LOW,
-      TOO_HIGH
-    };
+public class TypewiseAlert {
+
     public static BreachType inferBreach(double value, double lowerLimit, double upperLimit) {
-      if(value < lowerLimit) {
-        return BreachType.TOO_LOW;
-      }
-      if(value > upperLimit) {
-        return BreachType.TOO_HIGH;
-      }
-      return BreachType.NORMAL;
-    }
-    public enum CoolingType {
-      PASSIVE_COOLING,
-      HI_ACTIVE_COOLING,
-      MED_ACTIVE_COOLING
-    };
-    public static BreachType classifyTemperatureBreach(
-        CoolingType coolingType, double temperatureInC) {
-      int lowerLimit = 0;
-      int upperLimit = 0;
-      switch(coolingType) {
-        case PASSIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 35;
-          break;
-        case HI_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 45;
-          break;
-        case MED_ACTIVE_COOLING:
-          lowerLimit = 0;
-          upperLimit = 40;
-          break;
-      }
-      return inferBreach(temperatureInC, lowerLimit, upperLimit);
+        if (value < lowerLimit) {
+            return BreachType.TOO_LOW;
+        }
+        if (value > upperLimit) {
+            return BreachType.TOO_HIGH;
+        }
+        return BreachType.NORMAL;
     }
-    public enum AlertTarget{
-      TO_CONTROLLER,
-      TO_EMAIL
-    };
-    public class BatteryCharacter {
-      public CoolingType coolingType;
-      public String brand;
+
+    public static boolean checkAndAlert(
+            AlertTarget alertTarget, CoolingType coolingType, double temperatureInC) {
+
+        BreachType breachType = inferBreach(
+                temperatureInC, coolingType.getLowerLt(), coolingType.getUpperLt());
+
+        return alertForBreach(alertTarget, breachType);
     }
-    public static void checkAndAlert(
-        AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC) {
 
-      BreachType breachType = classifyTemperatureBreach(
-        batteryChar.coolingType, temperatureInC
-      );
+    private static boolean alertForBreach(AlertTarget alertTarget, BreachType breachType) {
 
-      switch(alertTarget) {
-        case TO_CONTROLLER:
-          sendToController(breachType);
-          break;
-        case TO_EMAIL:
-          sendToEmail(breachType);
-          break;
-      }
+        switch (alertTarget) {
```

Comment:
this switch case can be replaced by single method call, where the method could be an interface method, and either email target or controller target is an instance.
By abstracting it into an interface, the production code is made free of knowledge of which alert target needs to be called.